input/drivers/skeleton.py

```diff
@@ -23,16 +23,39 @@ def __init__(self, mapping=None, threaded=True):
             self.mapping = self.default_mapping
         try:
             self.init_hw()
-        except Exception as e:
-            logger.error("init_hw function not found!")
-            logger.exception(e)
+        except AttributeError:
+            logger.error("{}: init_hw function not found!".format(self.__class__))
+        try:
+            self.set_available_keys()
```

Comment:
This will never fail with an AttributeError - `set_available_keys` exists in this base class, so this call can only fail with ValueError. However, we need some kind of way for the HID driver to set `available_keys` to None, but avoid exceptions... I think I'll just override `set_available_keys` in HID driver to do it.

Reply:
...Oh wait, the HID driver already is taken into account!